src/CLR/Core/Serialization/BinaryFormatter.cpp

```
@@ -1041,16 +1043,16 @@ HRESULT CLR_RT_BinaryFormatter::TypeHandler::ReadValue(int &res)
             }
             else
             {
                 CLR_RT_HeapBlock_String *str = CLR_RT_HeapBlock_String::CreateInstance(*m_value, len);
                 CHECK_ALLOCATION(str);
-                char *dst1 = (char *)str->StringText();
+                auto dstString = (char *)str->StringText();
 
-                NANOCLR_CHECK_HRESULT(m_bf->ReadArray((CLR_UINT8 *)dst1, len));
-                dst1[len] = 0;
+                NANOCLR_CHECK_HRESULT(m_bf->ReadArray((CLR_UINT8 *)dstString, len));
+                dst[len] = 0;
```

Cppcheck (2.10-2)

[error] Null pointer dereference

(nullPointer)